### PR TITLE
eclipse-mat: init at 1.10.0.20200225

### LINKED ISSUE #43611: needs packaging: Eclipse Memory Analyzer
## Issue description
Eclipse Memory Analyzer is missing from nixpkgs. 
https://www.eclipse.org/mat/downloads.php

## Technical details

 - system: `"x86_64-linux"`
 - host os: `Linux 4.14.48, NixOS, 18.03.git.411cc55 (Impala)`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.0.4`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos/nixpkgs`

### PR DESCRIPTION
**Pull request by ktor**
Closes #43611

###### Motivation for this change
To add best, free, java memory analyzer to NixOS package collection.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by ktor** [comment]
Hi @rycee, could you please review the pull request? I've based the package on build-eclipse.nix expression from eclipse package because it was easier for me. I am not able to judge it would work within the eclipse package better and don't know how would I integrate it there. Could you please comment on that?

File: pkgs/development/tools/eclipse-mat/default.nix [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    libXrender libXtst makeWrapper zlib unzip shared-mime-info webkitgtk
+  ];
+
+  dontBuild = true;
+  dontConfigure = true;
+
+  meta = with stdenv.lib; {
+    description = "The Eclipse Memory Analyzer is a fast and feature-rich Java heap analyzer that helps you find memory leaks and reduce memory consumption";
+    longDescription = ''
+        Use the Memory Analyzer to analyze productive heap dumps with hundreds of millions of objects, quickly calculate the retained sizes of objects, see who is preventing the Garbage Collector from collecting objects, run a report to automatically extract leak suspects.
```
Comment (rycee) [inline pkgs/development/tools/eclipse-mat/default.nix:76]: Avoid having too long lines. Usually 80 characters is fine.

File: pkgs/development/tools/eclipse-mat/default.nix [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+  buildInputs = [
+    fontconfig freetype glib gsettings-desktop-schemas gtk3 jdk libX11
+    libXrender libXtst makeWrapper zlib unzip shared-mime-info webkitgtk
+  ];
+
+  dontBuild = true;
+  dontConfigure = true;
+
+  meta = with stdenv.lib; {
+    description = "The Eclipse Memory Analyzer is a fast and feature-rich Java heap analyzer that helps you find memory leaks and reduce memory consumption";
```
Comment (rycee) [inline pkgs/development/tools/eclipse-mat/default.nix:74]: The description should be short. See https://nixos.org/nixpkgs/manual/#sec-standard-meta-attributes.

File: pkgs/development/tools/eclipse-mat/default.nix [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    [ -f $libCairo ] && patchelf --set-rpath ${stdenv.lib.makeLibraryPath [ freetype fontconfig libX11 libXrender zlib ]} $libCairo
+
+    # Create wrapper script.  Pass -configuration to store settings in ~/.eclipse-mat/<version>
+    makeWrapper $out/mat/MemoryAnalyzer $out/bin/eclipse-mat \
+      --prefix PATH : ${jdk}/bin \
+      --prefix LD_LIBRARY_PATH : ${stdenv.lib.makeLibraryPath ([ glib gtk3 libXtst webkitgtk ])} \
+      --prefix XDG_DATA_DIRS : "$GSETTINGS_SCHEMAS_PATH" \
+      --add-flags "-configuration \$HOME/.eclipse-mat/''${version}/configuration"
+
+
```
Comment (rycee) [inline pkgs/development/tools/eclipse-mat/default.nix:56]:
```suggestion
```

**Comment by rycee** [comment]
This package does not seem to belong in the editors namespace so having it this way seems fine to me.

**Comment by ktor** [comment]
Thank you @rycee for the review! I've resolved all your suggestions.

**Comment by rycee** [comment]
Thanks! I ran nixpkgs-fmt to do some minor cleanups and rebased to master in 17e1fbe0fc683de504907c95253db55e1eacc685.

**Comment by ktor** [comment]
Thank you for the information @rycee , will use formatter in the future!

**Pull request closed without merging**